### ISSUE TITLE
Bad Request errors when uploading via API

### ISSUE THREAD

**Issue by pajot** [OpenEnergyPlatform/oeplatform#1086]
I have a URL that looks like this:

`url = 'https://postgres.monitoring:8000/api/v0/schema/model_draft/tables/test3/'`

I create a query using a function I wrote for this purpose, using a json string generated by a pandas dataframe:

```
data_empty = {'query': {'columns': [{'name': 'id',
    'data_type': 'bigserial',
    'is_nullable': 'NO'},
   {'name': 'CH_ID', 'data_type': 'integer'},
   {'name': 'DATE', 'data_type': 'text'},
   {'name': 'SINGLE_READING', 'data_type': 'integer'},
   {'name': 'VALUE', 'data_type': 'numeric'},
   {'name': 'MIN', 'data_type': 'text'},
   {'name': 'MAX', 'data_type': 'text'},
   {'name': 'SD', 'data_type': 'text'},
   {'name': 'VALID_SAMPLES', 'data_type': 'integer'}],
  'constraints': [{'constraint_type': 'PRIMARY KEY',
    'constraint_parameter': 'id'}]}}
```

I can create a table using this schema:
```
# Create table via API
requests.put(
    url,
    json=data_empty,
    headers={'Authorization': 'Token %s'%your_token}
)
<Response [201]>
```
I create a data dict that looks like this:
```
json_data_records = {'query': [{'CH_ID': 911,
   'DATE': '2018-03-01 17:30:00',
   'SINGLE_READING': 0,
   'VALUE': 86.6358,
   'MIN': '\\N',
   'MAX': '\\N',
   'SD': '\\N',
   'VALID_SAMPLES': 1},
  {'CH_ID': 911,
   'DATE': '2018-03-01 17:45:00',
   'SINGLE_READING': 0,
   'VALUE': 38.1173,
   'MIN': '\\N',
   'MAX': '\\N',
   'SD': '\\N',
   'VALID_SAMPLES': 1},
  {'CH_ID': 911,
   'DATE': '2018-03-01 18:00:00',
   'SINGLE_READING': 0,
   'VALUE': 7.43827,
   'MIN': '\\N',
   'MAX': '\\N',
   'SD': '\\N',
   'VALID_SAMPLES': 1}]}

```
But when I attempt to POST this with the following command, I get a 400 Bad Request error:
```
requests.post(
    url + 'rows/new',
    json = json_data_records,
    headers={'Authorization': 'Token %s'%your_token}
)
<Response [400]>
```

Can anyone help me troubleshoot this?

**Comment by pajot** [comment]
Okay, I captured the requests response object like so:
```
r = requests.post(
    url + 'rows/new',
    json = json_data_records,
    headers={'Authorization': 'Token %s'%your_token}
)
<Response [400]>
```
and now I have useful output:

`'{"reason": "CompileError(\'Unconsumed column names: MAX, SINGLE_READING, DATE, SD, MIN, VALID_SAMPLES, CH_ID, VALUE\')"}'`

It turns out that the table was not created correctly. In psql:

```
oep_db=# \dS model_draft.test3
                               Table "model_draft.test3"
 Column |  Type  | Collation | Nullable |                    Default
--------+--------+-----------+----------+-----------------------------------------------
 id     | bigint |           | not null | nextval('model_draft.test3_id_seq'::regclass)
Indexes:
    "test3_pkey" PRIMARY KEY, btree (id)
```
Why is this happening? I thought it might be the letter case of the column names, because the Create Table mask of the frontend rejects uppercase column names, but PostgreSQL doesn't care and folds them all down anyway.

Perhaps it's better to filter for lowercase, but the API shouldn't silently ignore column names with the wrong case. I had no idea there was a problem until I tried to upload records.

**Comment by pajot** [comment]
I have to apologize now. I did not create the table with the query I thought I had; the query I actually used _was_ empty. Arghhh!

When I use the correct query, the table is created properly without errors and the multiple records can be uploaded.

That's what I get for hacking in a Jupyter Notebook!

Thanks for putting up with my abuse of the issues applet to do [rubber duck debugging](https://en.wikipedia.org/wiki/Rubber_duck_debugging) ;-)
